packages/experiment-codec/tests/utils.test.ts

```diff
@@ -1,12 +1,38 @@
 import test from "ava";
-import { assertHexDecimal, assertHexString } from "../src/utils";
+import {
+  assertHexDecimal,
+  assertHexString,
+  isObjectLike,
+  toArrayBuffer,
+} from "../src/utils";
 
 test("should assertHexString fail", (t) => {
   t.throws(() => assertHexString("00"));
+  t.throws(() => assertHexString("0x01020304", 8));
+});
+test("should assertHexString pass", (t) => {
+  t.notThrows(() => assertHexString("0x01020304", 4));
 });
 test("should assertHexDecimal fail", (t) => {
   t.throws(() => assertHexDecimal("0x"));
   t.throws(() => assertHexDecimal("0x123", 1));
   t.notThrows(() => assertHexDecimal("0x0"));
   t.notThrows(() => assertHexDecimal("0x12", 1));
 });
+test("should toArray buffer return expected", (t) => {
```

Comment:
test("should return the expected ArrayBuffer when calling toArrayBuffer", (t) => {